heat/core/factories.py

```diff
@@ -289,6 +289,10 @@ def array(
     if dtype is not None:
         dtype = types.canonical_heat_type(dtype)
 
+    # sanitize device
+    if device is not None:
+        device = devices.sanitize_device(device)
+
```

Comment:
This is a good idea. It took my a while before I found the sanitation functions in stride_tricks :sweat_smile: 
I think it may be better for the next major release or in a seperate Pull Request.